cylc/flow/id_match.py

```diff
@@ -213,7 +215,7 @@ def filter_ids(
             if warn:
                 LOG.warning(f"No active tasks matching: {id_}")
         else:
-            _cycles.extend(cycles)
+            _cycles.extend(list(cycles))
```

Comment:
The intermediate list is not necessary:

```suggestion
            _cycles.extend(cycles)
```